pypesto/ensemble/ensemble.py

```diff
@@ -599,9 +605,19 @@ def from_sample(
                 geweke_test(result)
             burn_in = result.sample_result.burn_in
             x_vectors = x_vectors[burn_in:]
+        else:
+            burn_in = 0
+
+        # added cutoff
+        if rel_cutoff is None:
+            pass
+        else:
+            x_vectors = calculate_hpd(result=result, burn_in=burn_in, ci_level=rel_cutoff)
+
         if chain_slice is not None:
             x_vectors = x_vectors[chain_slice]
```

Comment:
A general remark, but it is not quite clear whether this is the desired order, it might be that we want to first slice and then calculate hpd. But that is only a minor thing here i guess.

Reply:
In principle, I would rather calculate the HPD first and then take a slice, as the HPD should become more precise that way; computationally, it's not an issue. But if there are other reasons to change the order, that would also be fine with me.